### PR TITLE
fix: include type definitions generated from JSDoc

### PR DESCRIPTION
**Pull request by wkillerud**
- Update devDependencies and use our shared configs
- Document the API with JSDoc and generate type definitions
- Update the README with examples from JSDoc

### CODE REVIEW

**Comment by wkillerud** [comment]
Windows fails because of https://github.com/eik-lib/common/pull/303

File: README.md [hunk 1]
```diff
@@ -1,191 +1,264 @@
 # @eik/node-client
 
-The Eik Node.js client facilitates loading Eik config and providing URLs to assets on an Eik server or in local development plus loading import maps from the Eik server.
+This is an Eik utility for servers running on Node. With it you can:
```
Comment (leftieFriele) [inline README.md:3]:
```suggestion
This is an Eik utility for working with [@eikservice](https://github.com/eik-lib/service) and your configuration. With it you can:
```
This feels a bit vague, as there is no other option than to run an Eik server on node using the `service` module?

File: src/index.js [hunk 2]
```diff
@@ -1,106 +1,297 @@
...
+ *        ${script.integrity ? `integrity="${script.integrity}"` : ""}
+ *        type="module"></script>
+ *    </body></html>`);
+ * });
+ *
+ * app.listen({
+ *    port: 3000,
+ * });
+ *
+ * console.log("Listening on http://localhost:3000");
```
Comment (leftieFriele) [inline src/index.js:104]: I was curious as to why repeat the readme docs here, as I would think one place would suffice and be easier to keep up to date over time 🤔

File: src/index.js [hunk 3]
```diff
@@ -1,106 +1,297 @@
...
+	 * The `"server"` field from the Eik config
+	 * @throws if read before calling {@link load}
+	 */
+	get server() {
+		if (this.#config.server) return this.#config.server;
+		throw new Error("Eik config was not loaded before calling .server");
+	}
+
+	/**
+	 * The pathname to the base on Eik (ex. /pkg/my-app/1.0.0/)
```
Comment (leftieFriele) [inline src/index.js:184]: "to the base on Eik" does that mean the same as "to the base path in the current sink"?

File: README.md [hunk 4]
```diff
@@ -1,191 +1,264 @@
...
-## Description
-
-This module will load Eik config from either an `eik.json` file or from values set in `package.json` and then use these values to provide absolute URLs to assets on a Eik server. In addition to this it's possible to set a `base` URL which will be used as the "base root" for files when this module is set in development mode. This makes it easy to retrieve absolute URLs to assets on a Eik server when an application is running in production but also get URLs to the same assets when in development.
+## Usage
 
-In addition this module can also download the import maps defined in Eik config and provide these for inclusion in an application.
+The most common use case for this module is linking to a file.
 
-The following will use the information in Eik config and provide an absolute URL to a file on an Eik server:
+When developing you typically want to use a local version of the file, then link to the published version on Eik when running in production.
```
Comment (leftieFriele) [inline README.md:18]: This key difference between development and production I think could be emphasised a bit more, as having the client in development mode means it does something entirely different than in production 🤔 

I see it mentioned in the method docs, which is really good, I was just thinking for my self I would read this passage and just glance over the method docs. Maybe we could write something about this difference in behaviour here too.

**Review by leftieFriele — APPROVED** [review]
This is a really great job tidying up all the configs + deps, while improving the documentation by 💯 . 🌮  for you.

File: src/index.js (hunk 2 above)
Comment (wkillerud) [reply src/index.js:104]: Certainly easier to update, but I think there's value in having the docs both show up in an editor and available on the web. Honestly I'm thinking about adding a third location – the docs on https://eik.dev 😅 (For MetricsJS I ended up [moving much of the README.md to be only on the docsite](https://github.com/metrics-js/client#readme). Maybe we do something like that?)

File: src/index.js (hunk 3 above)
Comment (wkillerud) [reply src/index.js:184]: No, the base path of the sink would be closer to `/`. I mean for example `/pkg/my-app/1.0.0/`.

Comment (leftieFriele) [reply src/index.js:184]: riiight, so it's the base path for the current package.

**Pull request merged**